### ISSUE TITLE
[Bug]: [Nightly]Compact timeout reporting waitting new segment index and compact more than 240s

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#20595]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 221d044
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): 2.2.0.dev71
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Compact timeout reporting waitting new segment index and compact more than 240s.
```
[2022-11-14T21:16:31.732Z]     @pytest.mark.tags(CaseLabel.L2)
[2022-11-14T21:16:31.732Z]     def test_compact_during_insert(self):
[2022-11-14T21:16:31.732Z]         """
[2022-11-14T21:16:31.732Z]         target: test compact during insert and flush
[2022-11-14T21:16:31.732Z]         method: 1.insert entities into multi segments
[2022-11-14T21:16:31.732Z]                 2.start a thread to load and search
[2022-11-14T21:16:31.732Z]                 3.compact collection
[2022-11-14T21:16:31.732Z]         expected: Search and compact both successfully
[2022-11-14T21:16:31.732Z]         """
[2022-11-14T21:16:31.732Z]         collection_w = self.collection_insert_multi_segments_one_shard(prefix, nb_of_segment=ct.default_nb,
[2022-11-14T21:16:31.732Z]                                                                        is_dup=False)
[2022-11-14T21:16:31.732Z]         collection_w.create_index(ct.default_float_vec_field_name, ct.default_index)
[2022-11-14T21:16:31.732Z]         log.debug(collection_w.index())
[2022-11-14T21:16:31.732Z]         df = cf.gen_default_dataframe_data()
[2022-11-14T21:16:31.732Z]     
[2022-11-14T21:16:31.732Z]         def do_flush():
[2022-11-14T21:16:31.732Z]             collection_w.insert(df)
[2022-11-14T21:16:31.732Z]             log.debug(collection_w.num_entities)
[2022-11-14T21:16:31.732Z]     
[2022-11-14T21:16:31.732Z]         # compact during insert
[2022-11-14T21:16:31.732Z]         t = threading.Thread(target=do_flush, args=())
[2022-11-14T21:16:31.732Z]         t.start()
[2022-11-14T21:16:31.732Z]         collection_w.compact()
[2022-11-14T21:16:31.732Z]         collection_w.wait_for_compaction_completed()
[2022-11-14T21:16:31.732Z]         collection_w.get_compaction_plans()
[2022-11-14T21:16:31.732Z]     
[2022-11-14T21:16:31.732Z]         t.join()
[2022-11-14T21:16:31.732Z]         # waitting for new segment index and compact
[2022-11-14T21:16:31.732Z]         cost = 240
[2022-11-14T21:16:31.732Z]         start = time()
[2022-11-14T21:16:31.732Z]         while True:
[2022-11-14T21:16:31.732Z]             sleep(20)
[2022-11-14T21:16:31.732Z]             collection_w.load()
[2022-11-14T21:16:31.732Z]             # new segment compacted
[2022-11-14T21:16:31.732Z]             seg_info = self.utility_wrap.get_query_segment_info(collection_w.name)[0]
[2022-11-14T21:16:31.732Z]             if len(seg_info) == 1:
[2022-11-14T21:16:31.732Z]                 break
[2022-11-14T21:16:31.732Z]             end = time()
[2022-11-14T21:16:31.732Z]             collection_w.release()
[2022-11-14T21:16:31.732Z]             if end - start > cost:
[2022-11-14T21:16:31.732Z] >               raise MilvusException(1, f"Waitting new segment index and compact more than {cost}s")
[2022-11-14T21:16:31.732Z] E               pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=Waitting new segment index and compact more than 240s)>
[2022-11-14T21:16:31.732Z] 
[2022-11-14T21:16:31.732Z] testcases/test_compaction.py:1226: MilvusException
```

### Expected Behavior

PASS

### Steps To Reproduce

https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/2.2.0/2/pipeline/135/

### Milvus Log

1. log: [artifacts-milvus-standalone-nightly-2-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/2.2.0/2/artifact/artifacts-milvus-standalone-nightly-2-pymilvus-e2e-logs.tar.gz)
2. collection name: compact_S3gWtVny
3. failed time: [2022-11-14T20:08:36.359Z] [gw3] [ 25%] FAILED testcases/test_compaction.py::TestCompactionOperation::test_compact_during_insert

**Comment by weiliu1031** [comment]
when small segment is smaller than 4, automatically compact will not be triggered. so the final segment num will be 2.
we need to refine the test case!

**Comment by MrPresent-Han** [comment]
assign MrPresent-Han

**Comment by weiliu1031** [comment]
/assign @NicoYuan1986 
please verify on this

**Comment by NicoYuan1986** [comment]
> [test] Fix case compact while insert #20662

Fixed for master branch. I will close the issue when it is fixed for 2.2.0 branch too.

**Comment by NicoYuan1986** [comment]
Fixed.
